conference/cart.py

```diff
@@ -101,7 +100,7 @@ def cart_step2_pick_tickets(request, type_of_tickets):
                     request, "The discount code provided expired or is invalid"
                 )
 
-        if CartActions.buy_tickets in request.POST:
+        elif CartActions.buy_tickets in request.POST:
```

Comment:
Not familiar with the full logic here, but it seems that in this case the discount will not be applied if it is submitted at the same time as the tickets buy action.

Reply:
Right - that's how the current implementation works; the view is overloaded to serve two functions - you either calculate the discount and just return it to the frontend so that the user can see whether applying the discount code worked, or you proceed with buying the tickets. The `cart_action` is used to switch between one and the other.

Reply:
Sounds a bit brittle, but I guess if it worked in the past should be fine for now.